### PR TITLE
Set the 'visibility' of 'FlutterView' to the previous one in 'FragmentActivityAndFragment.onStart'

### PR DESCRIPTION
**Pull request by ColdPaleLight**
fix https://github.com/flutter/flutter/issues/105203

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [ ] All existing and new tests are passing.

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java [hunk 1]
```diff
@@ -599,7 +602,10 @@ void onStop() {
     // screen when unlocked. We can work around this by changing the visibility of FlutterView in
     // onStart and onStop.
     // See https://github.com/flutter/flutter/issues/93276
-    flutterView.setVisibility(View.GONE);
+    if (flutterView.getVisibility() != View.GONE) {
```
Comment (Nayuta403) [inline shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java:605]:
```suggestion
```
I wonder if this judgment can be removed, or `previousVisibility` needs to be assigned again in else.
So let's say this is the case：
* First time the flutterView is `View.VISIBLE`. When `onStop` is executed, `previousVisibility` will also become `View.VISIBLE`.
* Second time I change the flutterView to `View.GONE`. When `onStop` is executed, the `previousVisibility` will not be changed. 

If `onStart` is executed at this time,` previousVisibility` remains as before, causes the flutterView to be `View.VISIBLE`（Even though I've changed to `View.GONE`）

**Review by Nayuta403 — COMMENTED** [review]
👍🏻  with some ideas

File: shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java (hunk 1 above)
Comment (ColdPaleLight) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java:605]: Good catch! I'll remove it.

**Review by dnfield — APPROVED** [review]

File: shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java (hunk 1 above)
Comment (ColdPaleLight) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java]: Done

**Comment by chdo002** [comment]
[issues/105203](https://github.com/flutter/flutter/issues/105203)
still exits;
this merge doesn't fix it

**Pull request merged**